bindings/python/parsers/urdf/geometry.cpp

```diff
@@ -139,10 +139,20 @@ namespace pinocchio
       template <class ArgumentPackage>
       static PyObject* postcall(ArgumentPackage const& args_, PyObject* result)
       {
+        // If owner_arg exist, we run bp::return_internal_reference postcall
+        // result lifetime will be tied to owner_arg lifetime
         PyObject* patient = bp::detail::get_prev<owner_arg>::execute(args_, result);
         if (patient != Py_None)
           return bp::return_internal_reference<owner_arg>::postcall(args_, result);
-        return result;
+        // If owner_arg doesn't exist, then Python will have to manage the result lifetime
+        bp::extract<GeometryModel*> geom_model_extract(result);
+        if (geom_model_extract.check())
+        {
+          return bp::to_python_indirect<GeometryModel, bp::detail::make_owning_holder>()(geom_model_extract());
+        }
+        // If returned value is not a GeometryModel*, then raise an error
+        PyErr_SetString(PyExc_RuntimeError, "pinocchio::python::return_value_policy only work on GeometryModel* data type");
```

Comment:
"only work**s**"

Reply:
the rest looks right to me, `make_owning_holder` seems to create a `unique_ptr` so it shouldn't leak